addons/pkg-maven/common/src/main/java/org/commonjava/indy/pkg/maven/content/MavenMetadataGenerator.java

```diff
@@ -611,16 +599,13 @@ private Metadata generateGroupMetadata( final Group group, final List<ArtifactSt
 
     private void putToMetadataCache( StoreKey key, String toMergePath, Metadata meta )
     {
-        synchronized ( versionMetadataCache )
+        Map cacheMap = versionMetadataCache.get( key );
+        if ( cacheMap == null )
```

Comment:
This is a potential race, since two calls to this method that are slightly out of sync could result in two map instances being used. Can we switch this to use `computeIfAbsent()` too?